src/WebJobs.Script/Description/DotNet/DependencyHelper.cs

```diff
@@ -91,9 +91,19 @@ private static string GetResourceFileContents(string fileName)
                 .ToDictionary(a => a.Name, StringComparer.OrdinalIgnoreCase);
         }
 
-        internal static ExtensionRequirementsInfo GetExtensionRequirements()
+        internal static ExtensionRequirementsInfo GetExtensionRequirements(IEnvironment environment)
         {
-            string requirementsJson = GetResourceFileContents("extensionrequirements.json");
+            string requirementsJson = string.Empty;
+
+            if (environment.IsFlexConsumptionSku())
+            {
+                requirementsJson = GetResourceFileContents("extensionrequirements_FlexConsumption.json");
```

Comment:
Why are we special casing a SKU here?

We should have the feature support a configuration that may be defined by the environment, avoiding SKU specific logic.

Reply:
Can you elaborate on what sort of configuration do you mean here?